spec.md

```diff
@@ -1103,6 +1313,48 @@ If a `CreateVolume` operation times out, leaving the CO without an ID with which
 
 It is NOT REQUIRED for a controller plugin to implement the `LIST_VOLUMES` capability if it supports the `CREATE_DELETE_VOLUME` capability: the onus is upon the CO to take into consideration the full range of plugin capabilities before deciding how to proceed in the above scenario.
 
+##### `CreateSnapshot`, `DeleteSnapshot`, `ListSnapshots`
+
+The plugin-generated `snapshot_id` is a REQUIRED field for the `DeleteSnapshot` RPC, as opposed to the CO-generated snapshot `name` that is REQUIRED for the `CreateSnapshot` RPC: these fields MAY NOT contain the same value.
+Therefore the CO-generated snapshot `name` specified in `CreateSnapshot` SHALL NOT be used as a filter to query a snapshot by calling `ListSnapshots`.
+Instead the plugin-generated `snapshot_id` SHALL be used as a filter to query a snapshot by calling `ListSnapshots`.
+If a `CreateSnapshot` operation times out, leaving the CO without an ID with which to reference a snapshot, and the CO also decides that it no longer needs/wants the snapshot in question then the CO MAY choose one of the following paths:
+
+1. Execute the `ListSnapshots` RPC to possibly obtain a snapshot ID that may be used to execute a `DeleteSnapshot` RPC; upon success execute `DeleteSnapshot`.
+2. The CO takes no further action regarding the timed out RPC, a snapshot is possibly leaked and the operator/user is expected to clean up.
+
+It is NOT REQUIRED for a controller plugin to implement the `LIST_SNAPSHOTS` capability if it supports the CREATE_DELETE_SNAPSHOT capability: the CO needs to take into consideration the full range of plugin capabilities before deciding how to proceed in the above scenario.
+
+##### Snapshot Statuses
+
+A snapshot could have the following statusus: UPLOADING, READY, and ERROR.
+
+Some cloud providers will upload the snapshot to a location in the cloud (i.e., an object store) after the snapshot is cut.
+Uploading may be a long process that could take hours.
+If a `freeze` operation was done on the application before taking the snapshot, it could be a long time before the application can be running again if we wait until the upload is complete to `thaw` the application.
```

Comment:
this is good information. the organization feels a bit clumsy. i might take a stab at crafting a PR against this branch to reorganize this section for better "flow"

Reply:
Sure.  Thanks:).